src/components/Contexts/FormDataProvider.jsx

```diff
@@ -251,7 +250,7 @@ const FormDataProvider = ({ children }) => {
 
         Alerter.success('common.saveFile.success')
       } catch (error) {
-        console.error(error)
+        log('error', error)
         Alerter.error('common.saveFile.error')
       }
       setIsStepperDialogOpen(false)
```

Comment:
You should start working on this component: 
- move all the non JSX stuff into helpers like all of that https://github.com/cozy/mespapiers/blob/39ec05239831231bf27c33e4ba14e0193a8d80a0/src/components/Contexts/FormDataProvider.jsx#L22-L111 => You should also avoid mutating params. We should try yo create new object to avoid side effect. You should start adding tests for that 
- This saveToPDF method should not be in a JSX file https://github.com/cozy/mespapiers/blob/39ec05239831231bf27c33e4ba14e0193a8d80a0/src/components/Contexts/FormDataProvider.jsx#L113-L174 . You should move it and test it 
- Does the only difference between https://github.com/cozy/mespapiers/blob/39ec05239831231bf27c33e4ba14e0193a8d80a0/src/components/Contexts/FormDataProvider.jsx#L214-L248 is that we call sometime `await PDFDocument.create()` ? If yes, we can refactor a little bit to be more readable, right?

Reply:
Indeed, the refactoring of this component is planned, I think to do it after this PR